plugins/module_utils/foreman_helper.py

```diff
@@ -1210,7 +1207,7 @@ def _flatten_entity(entity, foreman_spec):
             spec = foreman_spec[key]
             flat_name = spec.get('flat_name', key)
             property_type = spec.get('type', 'str')
-            if property_type == 'entity':
+            if property_type in ['entity', 'invisible-entity']:
```

Comment:
I had to introduce an `invisible-entity` as `invisible` doesn't get the nice special handling like `entity`, but I also really wanted it invisible.

Reply:
So being invisible and being an entity are actually orthogonal properties?

Reply:
Yeah, it seems so.

Reply:
Is there a use for "invisible-str" or "invisible-dict"?

Reply:
Let's wait until there is a need for it. But the next time we touch this, we should probably have a boolean called `invisible` in the `foreman_spec`.

Reply:
I have thought the same (while having some :icecream:) and I think I would prefer implementing it now, rather than next time. I'll at least give it a shot tomorrow.